allennlp/data/dataset.py

```diff
@@ -77,7 +77,8 @@ def get_padding_lengths(self) -> Dict[str, Dict[str, int]]:
 
     def as_arrays(self,
                   padding_lengths: Dict[str, Dict[str, int]] = None,
-                  verbose: bool = True) -> Dict[str, List[numpy.array]]:
+                  verbose: bool = True) -> Dict[str, List[Union[numpy.array,
```

Comment:
Either way, though, I'm pretty sure the `List` shouldn't be there.  If we stick with the union type, it should just be `Dict[str, Union[numpy.array, Dict[str, numpy.array]]`.